src/plugin.js

```diff
@@ -364,66 +364,69 @@ var zoomPlugin = {
 		if (!options.zoom || !options.zoom.enabled) {
 			return;
 		}
-		if (options.zoom.drag) {
-			chartInstance.zoom._mouseDownHandler = function(event) {
+
+		chartInstance.zoom._mouseDownHandler = function(event) {
+			if (chartInstance.options.zoom.drag) {
 				chartInstance.zoom._dragZoomStart = event;
-			};
-			node.addEventListener('mousedown', chartInstance.zoom._mouseDownHandler);
+			}
+		};
+		node.addEventListener('mousedown', chartInstance.zoom._mouseDownHandler);
 
-			chartInstance.zoom._mouseMoveHandler = function(event) {
-				if (chartInstance.zoom._dragZoomStart) {
-					chartInstance.zoom._dragZoomEnd = event;
-					chartInstance.update(0);
+		chartInstance.zoom._mouseMoveHandler = function(event) {
+			if (chartInstance.options.zoom.drag && chartInstance.zoom._dragZoomStart) {
+				chartInstance.zoom._dragZoomEnd = event;
+				chartInstance.update(0);
+			}
+		};
+		node.addEventListener('mousemove', chartInstance.zoom._mouseMoveHandler);
```

Comment:
I forgot we have a `mousemove` handler. `mousemove` can fire hundreds of times a second constantly, so we should be a lot more careful than with `mouseup`, `mousedown`, etc. I'd be more comfortable for performance reasons if went the route of switching the listeners on and off instead eventhough it would be a bit more complicated.

Reply:
Good point! I fixed and now add the `mousemove` listener on `mousedown`. Other listeners stay initialized as it is.